wcfsetup/install/files/ts/WoltLabSuite/Core/Controller/Clipboard.ts

```diff
@@ -0,0 +1,706 @@
+/**
+ * Clipboard API Handler.
+ *
+ * @author  Alexander Ebert
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Controller/Clipboard
+ */
+
+import * as Ajax from "../Ajax";
+import { AjaxCallbackSetup } from "../Ajax/Data";
+import * as Core from "../Core";
+import DomChangeListener from "../Dom/Change/Listener";
+import DomUtil from "../Dom/Util";
+import * as EventHandler from "../Event/Handler";
+import * as Language from "../Language";
+import * as UiConfirmation from "../Ui/Confirmation";
+import UiDropdownSimple from "../Ui/Dropdown/Simple";
+import * as UiPageAction from "../Ui/Page/Action";
+import * as UiScreen from "../Ui/Screen";
+
+interface ClipboardOptions {
+  hasMarkedItems: boolean;
+  pageClassName: string;
+  pageObjectId: number;
+}
+
+interface ContainerData {
+  checkboxes: HTMLCollectionOf<HTMLInputElement>;
+  element: HTMLElement;
+  markAll: HTMLInputElement | null;
+  markedObjectIds: Set<number>;
+}
+
+interface ItemData {
+  items: { [key: string]: ClipboardActionData };
+  label: string;
+  reloadPageOnSuccess: string[];
+}
+
+interface ClipboardActionData {
+  actionName: string;
+  internalData: ArbitraryObject;
+  label: string;
+  parameters: {
+    actionName?: string;
+    className?: string;
+    objectIDs: number[];
+    template: string;
+  };
+  url: string;
+}
+
+interface AjaxResponseMarkedItems {
+  [key: string]: number[];
+}
+
+interface AjaxResponse {
+  actionName: string;
+  returnValues: {
+    action: string;
+    items?: {
+      // They key is the `typeName`
+      [key: string]: ItemData;
```

Comment:
Likewise this could be a record, possibly with a `type TypeName = string` as the key.